packages/authenticator/amplify_authenticator/lib/src/blocs/auth/auth_bloc.dart

```diff
@@ -333,6 +339,43 @@ class StateMachineBloc
             allowedMfaTypes: result.nextStep.allowedMfaTypes,
           ),
         );
+      case AuthSignInStep.continueSignInWithMfaSetupSelection:
```

Comment:
nit: instead of nesting the if statements I think we should use breaks.
```
if (allowedMfaTypes == null) {
    _emit(
        ContinueSignInWithMfaSetupSelection(
            allowedMfaTypes: result.nextStep.allowedMfaTypes,
        ),
    );
    break;
}

Reply:
+1, please refactore the nested if else statement. if they are mutually exclusive then no need to nest them.
for example mfaType == MfaType.totp, mfaType == MfaType.email are mutally exclusive so no need for if esle if statement